src/WidgetApi.ts

```diff
@@ -430,6 +434,52 @@ export class WidgetApi extends EventEmitter {
         ).then(r => r.events);
     }
 
+    /**
+     * Reads all related events given a known eventId.
+     * @param eventId The id of the parent event to be read.
+     * @param roomId The room to look within. When undefined, the user's currently
+     * viewed room.
+     * @param relationType The relationship type of child events to search for.
+     * When undefined, all relations are returned.
+     * @param eventType The event type of child events to search for. When undefined,
+     * all related events are returned.
+     * @param limit The maximum number of events to retrieve per room. If not
+     * supplied, the server will apply a default limit.
+     * @param from The pagination token to start returning results from, as
+     * received from a previous call. If not supplied, results start at the most
+     * recent topological event known to the server.
+     * @param to The pagination token to stop returning results at. If not
+     * supplied, results continue up to limit or until there are no more events.
+     * @param direction The direction to search for according to MSC3715.
+     * @returns Resolves to the room relations.
+     */
+    public readEventRelations(
+        eventId: string,
+        roomId?: string,
+        relationType?: string,
+        eventType?: string,
+        limit?: number,
+        from?: string,
+        to?: string,
+        direction?: 'f' | 'b',
+    ): Promise<IReadRelationsFromWidgetResponseData> {
```

Comment:
we have been doing a bad job of doing it, but on the request side we should be checking to see if the client supports the feature we're about to make a request to. This can be done in a couple ways:

* Copy `supportsMSC2974Renegotiate` in this class
* Call `if ((await this.getClientVersions()).includes(UnstableApiVersion.MSC3869))` at the start of this function.

I'm somewhat more fond of the second, personally, as it involves less caching/cleanup of variables. 

We'll also need to add the new UnstableApiVersion to `CurrentApiVersions` in `src/interfaces/ApiVersion.ts`.

Reply:
Would you prefer to throw an Error in that case or just return an "empty" response?

In general, we also thought if it might be useful if the `getClientVersions()` function would be public so a widget could actively check what is supported and adapt e.g. the UI to the available features.

Reply:
I think throwing an error makes sense, yeah. Making `getClientVersions` public also makes sense to me, though I think that should be done in another PR.